src/spu/src/services/public/stream_fetch.rs

```diff
@@ -331,53 +320,52 @@ impl StreamFetchHandler {
-                        records,
-                        next_filter_offset: next_offset,
-                        // we mark last offset in the response that we should sync up
-                        ..Default::default()
-                    };
+                debug!(
+                    next_offset,
+                    records = filter_batch.records().len(),
+                    "sending back to consumer"
+                );
+                let records = RecordSet::default().add(filter_batch);
+                let filter_partition_response = DefaultPartitionResponse {
+                    partition_index: self.replica.partition,
+                    error_code: file_partition_response.error_code,
+                    high_watermark: file_partition_response.high_watermark,
+                    log_start_offset: file_partition_response.log_start_offset,
+                    records,
+                    next_filter_offset: next_offset,
+                    // we mark last offset in the response that we should sync up
+                    ..Default::default()
+                };
 
-                    let filter_response = StreamFetchResponse {
-                        topic: self.replica.topic.clone(),
-                        stream_id: self.stream_id,
-                        partition: filter_partition_response,
-                    };
+                let filter_response = StreamFetchResponse {
+                    topic: self.replica.topic.clone(),
+                    stream_id: self.stream_id,
+                    partition: filter_partition_response,
+                };
 
-                    let filter_response_msg =
-                        RequestMessage::<DefaultStreamFetchRequest>::response_with_header(
-                            &self.header,
-                            filter_response,
-                        );
+                let filter_response_msg =
+                    RequestMessage::<DefaultStreamFetchRequest>::response_with_header(
+                        &self.header,
+                        filter_response,
+                    );
 
-                    trace!("sending back filter respone: {:#?}", filter_response_msg);
+                trace!("sending back filter respone: {:#?}", filter_response_msg);
 
-                    let mut inner_sink = self.sink.lock().await;
-                    inner_sink
-                        .send_response(&filter_response_msg, self.header.api_version())
-                        .await?;
-                    true
-                } else {
-                    debug!(next_offset, "filter, no records send back, skipping");
-                    false
-                };
+                let mut inner_sink = self.sink.lock().await;
+                inner_sink
+                    .send_response(&filter_response_msg, self.header.api_version())
+                    .await?;
 
-                Ok((next_offset, consumer_wait))
+                Ok((next_offset, true))
```

Comment:
it's hard to follow change here.  let's revert this sections

Reply:
Reverted. The original reason I put this here is because up above, we check if `consumer_wait` is `false` and return if so, meaning that if we get to this point in the code, `consumer_wait` must be `true`. But it doesn't make a difference if we keep using the variable.